docs/csharp/misc/cs0219.md

```diff
@@ -1,27 +1,40 @@
 ---
 title: "Compiler Warning (level 3) CS0219"
-ms.date: 07/20/2015
+ms.date: 09/06/2019
 f1_keywords: 
   - "CS0219"
 helpviewer_keywords: 
   - "CS0219"
 ms.assetid: 7945c497-4bfa-4695-9166-815a2ad0c8e7
 ---
 # Compiler Warning (level 3) CS0219
-The variable 'variable' is assigned but its value is never used  
-  
- The compiler issues a level-three warning, when you declare and assign a variable, but do not use it.  
-  
- The following sample generates CS0219:  
-  
-```csharp  
-// CS0219.cs  
-// compile with: /W:3  
-public class MyClass  
-{  
-   public static void Main()  
-   {  
-      int a = 0;   // CS0219  
-   }  
+The variable 'variable' is assigned but its value is never used
+
+ The compiler issues a level-three warning, when you declare and assign a variable, but do not use it.
+
+ > [!NOTE]
+ > The compiler generates this warning only when the variable value is a compile-time constant. Because interpolated strings are transformed into a <xref:System.String.Format%2A?displayProperty=nameWithType> method call, it's not considered a compile-time constant.
```

Comment:
I think a better note would reference the justification in dotnet/roslyn#15695, namely that assigning the result of some expression to an (otherwise) unused local variable keeps that in memory and enables viewing with a debugger.

```suggestion
 > The compiler generates this warning only when the variable value is a compile-time constant. Assigning an expression or method result to a local variable makes it easier to observe those expressions in the debugger. It also makes the result reachable, preventing garbage collection while that variable is reachable.
```

Reply:
I just noticed that `Assigning an expression or method result` seems to be in-accurate I think.
because regular string literal or number like `"string"` or `0` in:

```csharp
var x = "string";
var y = 0;
```

can be considered as expressions. Actually, there are the simplest expressions. But... assigning them to an unused variable generates the warning, while assigning a method result doesn't.

Reply:
Good point. I updated the suggestion.

Reply:
@BillWagner , The updated suggestion looks good. I've applied it.